src/Client/ClientBase.cpp

```diff
@@ -2497,6 +2550,11 @@ void ClientBase::init(int argc, char ** argv)
 
     readArguments(argc, argv, common_arguments, external_tables_arguments, hosts_and_ports_arguments);
 
+    /// Support for Unicode dashes
+    /// Interpret Unicode dash as default double-dash
+    for (auto & arg : common_arguments)
+        arg = std::regex_replace(arg, std::regex(UNICODE_DASH), "--");
```

Comment:
Regular expressions are unneeded here.

(+ whenever I see C++ standard regex I shudder, because C++ standard regexps are bad)

Reply:
fixed
changed to boost::replace_all